### DISCUSSION TITLE
Is this a webpack bug ? What should be changed?

### DISCUSSION THREAD

**Discussion by ghost** [webpack/webpack#16654] [General]
```ts
import meow from 'meow';
export const cliOptions = meow(
  {
    importMeta: import.meta,
   //...
}).flags;

export type CliOptions = typeof cliOptions;
```

<details>
<summary>
This is the config
</summary

```
const path = require('node:path');
const { fileURLToPath } = require('url');

module.exports = {
  mode: 'production',
  target: 'node',
  resolve: {
    extensions: ['.js', '.ts'],
    // Add support for TypeScripts fully qualified ESM imports.
    extensionAlias: {
      '.js': ['.js', '.ts'],
    },
  },
  output: {
    path: path.resolve(__dirname, 'bin'),
    filename: 'index.js',
  },
  entry: './src/index.ts',
  module: {
    rules: [{ test: /\.ts$/, use: 'ts-loader' }],
  },
};

```

</details>

It errors out at compiling:

> WARNING in ./src/cliOptions.ts 22:16-27
> Critical dependency: Accessing import.meta directly is unsupported (only property access is supported)
>    at  .../node_modules/webpack/lib/dependencies/ImportMetaPlugin.js:85:10

Any pointers would be so much appreciated.

**Comment by chenxsan** [comment]
I think you can't access `import.meta` directly, but you should be able to access its properties like `import.meta.url`.

**Reply by ghost** [reply]
Yes but the package explicitly sets ImportMeta as the type. I guess I'll try to overwrite it..

**Reply by alexander-akait** [reply]
I think we should provide empty object for such cases or with `{ url: "pseudo-import.url" }`

**Reply by alexander-akait** [reply]
Can you open an issue?

**Reply by ghost** [reply]
Yes tomorrow will do @alexander-akait
